File: code/cpp/src/pta/done/ladder/p2_035.cpp
#include <iostream>
using namespace std;

const int N_MAX = 30;

int n, tree[N_MAX];

void dfs(int index) {
    if (n <= index)
        return;
    dfs(2 * index + 1);
    dfs(2 * index + 2);
    cin >> tree[index];
}

int main() {
    cin >> n;
    dfs(0);
    cout << tree[0];
    for (int i = 1; i < n; i += 1) {
        cout << ' ' << tree[i];
    }
    cout << endl;
    return 0;
}

File: code/py/src/pta/done/ladder/p2_035.py
def main() -> None:
    n = int(input())
    nums = map(int, input().split())

    tree = [0 for _ in range(n)]

    def dfs(index: int) -> None:
        if n <= index:
            return
        dfs(index = 2 * index + 1)
        dfs(index = 2 * index + 2)
        tree[index] = next(nums)

    dfs(index = 0)
    print(*tree)


if __name__ == '__main__':
    main()
